lib/ansible/executor/module_common.py

```diff
@@ -304,11 +304,15 @@ def debug(command, zipped_mod, json_params):
     ANSIBALLZ_PARAMS = %(params)s
     if PY3:
         ANSIBALLZ_PARAMS = ANSIBALLZ_PARAMS.encode('utf-8')
+    remove_tempdir = False
     try:
-        # There's a race condition with the controller removing the
-        # remote_tmpdir and this module executing under async.  So we cannot
-        # store this in remote_tmpdir (use system tempdir instead)
-        temp_path = tempfile.mkdtemp(prefix='ansible_')
+        # this temp path will be under the 'remote_tmp', in it's own subdir
+        tempdir = os.environ.get('ANSIBLE_REMOTE_TEMP', None) or None
```

Comment:
Why do we need to do this instead of just substituting the temp dir into the python string?

Reply:
Hmm... Because you want to be able to set it differently per-host.... Okay.

Reply:
can be done, but since i need the env var for other cases (async_wrapper) I stuck with that